Content.Server/Atmos/EntitySystems/GasTileOverlaySystem.cs

```diff
@@ -145,9 +142,9 @@ private void OnPlayerStatusChanged(object? sender, SessionStatusEventArgs e)
         /// <param name="indices"></param>
         /// <param name="overlayData"></param>
         /// <returns>true if updated</returns>
-        private bool TryRefreshTile(GridId grid, GasOverlayData oldTile, Vector2i indices, out GasOverlayData overlayData)
+        private bool TryRefreshTile(GridAtmosphereComponent gridAtmosphere, GasOverlayData oldTile, Vector2i indices, out GasOverlayData overlayData)
         {
-            var tile = _atmosphereSystem.GetTileAtmosphereOrCreateSpace(grid, indices);
+            var tile = _atmosphereSystem.GetTileAtmosphere(gridAtmosphere, indices);
 
             if (tile == null)
```

Comment:
So previously this never returned null, despite there being an explicit null check.
AFAICT It works fine if the return value is allowed to be null, but if this somehow causes issues this could be reverted to the previous behaviour (which is basically to just to output the same `default GasOverlayData`, but sometimes with a `true` return value).

Reply:
Whole thing needs refactoring anyway.